src/halmos/__main__.py

```diff
@@ -29,6 +29,7 @@ def parse_args(args) -> argparse.Namespace:
     parser.add_argument('--function', metavar='FUNCTION_NAME_PREFIX', default='check', help='run tests matching the given prefix only (default: %(default)s)')
 
     parser.add_argument('--bytecode', metavar='HEX_STRING', help='execute the given bytecode')
+    parser.add_argument('--reset-bytecode', metavar='ADDR1=CODE1,ADDR2=CODE2,...', help='reset the bytecode of given addresses after setUp()')
```

Comment:
we wouldn't expect this to be used by "regular" end-users, right? 

maybe we can create an `advanced` or `internal` argument group:

```py
group_internal = parser.add_argument_group("Internal options")
group_internal.add_argument('--reset-bytecode', metavar='ADDR1=CODE1,ADDR2=CODE2,...', help='reset the bytecode of given addresses after setUp()')
```

Reply:
sounds great, now i divided options to multiple groups. now it looks like:
```
$ halmos -h
usage: halmos [-h] [--contract CONTRACT_NAME] [--function FUNCTION_NAME_PREFIX] [--loop MAX_BOUND] [--width MAX_WIDTH] [--depth MAX_DEPTH] [--array-lengths NAME1=LENGTH1,NAME2=LENGTH2,...] [--symbolic-storage] [-v] [--debug] [--log LOG_FILE_PATH] [--print-revert]
              [--help-compile] [--no-smt-add] [--no-smt-sub] [--no-smt-mul] [--smt-div] [--smt-div-by-const] [--smt-mod-by-const] [--smt-exp-by-const N] [--solver-timeout-branching TIMEOUT] [--solver-timeout-assertion TIMEOUT] [--solver-fresh] [--solver-subprocess]
              [--bytecode HEX_STRING] [--reset-bytecode ADDR1=CODE1,ADDR2=CODE2,...] [--symbolic-jump] [--print-potential-counterexample]
              [TARGET_DIRECTORY]

positional arguments:
  TARGET_DIRECTORY      source root directory (default: current directory)

options:
  -h, --help            show this help message and exit
  --contract CONTRACT_NAME
                        run tests in the given contract only
  --function FUNCTION_NAME_PREFIX
                        run tests matching the given prefix only (default: check)
  --loop MAX_BOUND      set loop unrolling bounds (default: 2)
  --width MAX_WIDTH     set the max number of paths
  --depth MAX_DEPTH     set the max path length
  --array-lengths NAME1=LENGTH1,NAME2=LENGTH2,...
                        set the length of dynamic-sized arrays including bytes and string (default: loop unrolling bound)
  --symbolic-storage    set default storage values to symbolic

Debugging options:
  -v, --verbose         increase verbosity levels: -v, -vv, -vvv, -vvvv
  --debug               run in debug mode
  --log LOG_FILE_PATH   log individual execution steps in JSON
  --print-revert        print reverting paths in verbose mode

Build options:
  --help-compile        print build options (foundry, hardhat, etc.)

Solver options:
  --no-smt-add          do not interpret `+`
  --no-smt-sub          do not interpret `-`
  --no-smt-mul          do not interpret `*`
  --smt-div             interpret `/`
  --smt-div-by-const    interpret division by constant
  --smt-mod-by-const    interpret constant modulo
  --smt-exp-by-const N  interpret constant power up to N (default: 2)
  --solver-timeout-branching TIMEOUT
                        set timeout (in milliseconds) for solving branching conditions (default: 1000)
  --solver-timeout-assertion TIMEOUT
                        set timeout (in milliseconds) for solving assertion violation conditions (default: 60000)
  --solver-fresh        run an extra solver with a fresh state for unknown
  --solver-subprocess   run an extra solver in subprocess for unknown

Internal options:
  --bytecode HEX_STRING
                        execute the given bytecode
  --reset-bytecode ADDR1=CODE1,ADDR2=CODE2,...
                        reset the bytecode of given addresses after setUp()

Experimental options:
  --symbolic-jump       support symbolic jump destination
  --print-potential-counterexample
                        print potentially invalid counterexamples

For more information, see https://github.com/a16z/halmos
```